### PR TITLE
Implemented aliases based on command line parameters.

### PR DESCRIPTION
**Pull request by Owlblocks**
This should implement the generic solution to #36 's alias addition. It uses command line parameters, so it can get pretty wordy. If using a config library would be preferable, [https://github.com/dotnetconfig/dotnet-config](url) appears to be something that would work.

### CODE REVIEW

File: src/GetText.Extractor/Program.cs [hunk 1]
```diff
@@ -25,27 +26,46 @@ private static async Task<int> Main(string[] args)
...
             Option sort = CommandLineOptions.SortOutput;
+            Option aliasString = CommandLineOptions.GetStringAliases;
+            Option aliasDefinite = CommandLineOptions.GetParticularStringAliases;
+            Option aliasPlural = CommandLineOptions.GetPluralStringAliases;
+            Option aliasDefinitePlural = CommandLineOptions.GetParticularPluralStringAliases;
+
+            aliasString.AllowMultipleArgumentsPerToken = true;
+            aliasDefinite.AllowMultipleArgumentsPerToken = true;
+            aliasPlural.AllowMultipleArgumentsPerToken = true;
+            aliasDefinitePlural.AllowMultipleArgumentsPerToken = true;
```
Comment (perpetualKid) [inline src/GetText.Extractor/Program.cs:37]: can we move these where the CommandLineOptions for aliases are statically defined:

```suggestion
internal static Option<List<string>> GetStringAliases => new Option<List<string>>(new[] { "--aliasgetstring", "-as" }, "List of aliases for GetString") { AllowMultipleArgumentsPerToken = true}; //etc for the other alias options

File: src/GetText.Extractor/Program.cs [hunk 2]
```diff
@@ -25,27 +26,46 @@ private static async Task<int> Main(string[] args)
...
+            List<string> getStringAliases, List<string> getParticularStringAliases, List<string> getPluralStringAliases, List<string> getParticularPluralStringAliases)
         {
             catalog = new CatalogTemplate(target.FullName);
 
-            SyntaxTreeParser parser = new SyntaxTreeParser(catalog, sources, unixStyle, verbose);
+            SyntaxTreeParser parser = new SyntaxTreeParser(catalog, sources, unixStyle, verbose, new Aliases()
+            {   GetString = new List<string>() { "GetString" }.Concat(getStringAliases).ToList(),
+                GetParticularString = new List<string>() { "GetParticularString" }.Concat(getParticularStringAliases).ToList(),
+                GetPluralString = new List<string>() { "GetPluralString" }.Concat(getPluralStringAliases).ToList(),
+                GetParticularPluralString = new List<string>() { "GetParticularPluralString" }.Concat(getParticularPluralStringAliases).ToList() });
```
Comment (perpetualKid) [inline src/GetText.Extractor/Program.cs:68]: please keep the default aliases out of program and where they were (ParserBase), such as
```suggestion
 in Engine\ParserBase.cs
        public ParserBase(CatalogTemplate catalog, bool unixStyle, bool verbose, Aliases aliases)
        {
.....
            GetStringAliases = new List<string>() {"GetString"}.AddRange(aliases.GetString ?? Enumerable.Empty<string>()).ToHashSet(); // etc for other aliases

**Review by perpetualKid — CHANGES_REQUESTED** [review]
looks good, with some minor comments which are more cosmetically, if you could please address these, we can merge and publish a new (preview) version! 
Thanks for contribution!

File: src/GetText.Extractor/Program.cs (hunk 2 above)
Comment (Owlblocks) [reply src/GetText.Extractor/Program.cs]: The only difference was that AddRange doesn't work in that syntax, as it modifies the list and returns nothing. Other than that, I basically copied your suggestions exactly.

**Pull request merged**